frontend/src/components/Header/TitleBanner/SearchBar.js

```diff
@@ -1,14 +1,26 @@
 import { BiSearch } from "react-icons/bi"
+import { useRef } from "react"
+import { useNavigate } from "react-router-dom"
 
 const SearchBar = () => {
   const searchIcon = <BiSearch className="text-2xl" />
-  
+  const formRef = useRef()
+  const navigate = useNavigate();
+
+  const handleSearch = (e) => {
+    e.preventDefault()
+    const formData = new FormData(formRef.current)
+    const data = Object.fromEntries(formData)
+    navigate(`/app/search?query=${encodeURIComponent(data.searchTerm)}`)
+    e.target.reset()
+  }
+
   return (
-    <div className="flex justify-between border-2 rounded-md border-gray-600 mx-2 px-1 hover:text-indigo-400/80 hover:border-indigo-400/80 ease-in duration-150">
-      <input type="text" className="bg-transparent focus:outline-none w-full"></input>
-      <button>{searchIcon}</button>
-    </div>
+    <form ref={formRef} onSubmit={handleSearch} className="flex justify-between border-2 rounded-md border-gray-600 mx-2 px-1 hover:text-indigo-400/80 hover:border-indigo-400/80 ease-in duration-150">
+      <input type="text" name="searchTerm" className="bg-transparent focus:outline-none w-full"></input>
+      <button type="submit">{searchIcon}</button>
+    </form>
   )
 }
 
-export default SearchBar;
\ No newline at end of file
+export default SearchBar
```

Comment:
If we press enter on an empty search bar or click the search icon, the search page displays all of our available products. Is this the normal behavior we should anticipate? Should we add a frontend block that requires something to be typed in before the form can be submitted or display a response that says please try again - can't search an empty term?

Reply:
Nice catch that is a bug for sure. The search query is `%search_term%` and that means that it would pull all data because the wildcard % characters are the only ones present.
A fix for this won't be super simple because it uncovered another bug. GraphQL isn't like REST and will always return 200s regardless if we raise an error or not. We're raising the error in the `search_products.rb` but that does nothing because it's not being handled anywhere. Mutations are being handled but queries are not. I'll have to edit some of these hooks to fix some queries.